From ad79a14fd384de7572b3e570672a4b7752b4d033 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:18:02 -0400
Subject: [alsa] hda: Move HDA_FIXUP_ACT_FREE call in snd_hda_gen_free()

Message-id: <140972823573-96-patchset-jkysela@redhat.com>
Patchwork-id: 90796
O-Subject: [RHEL 7.1 PATCH 095/442] ALSA: hda - Move HDA_FIXUP_ACT_FREE call in snd_hda_gen_free()
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Now Realtek and Conexant codec parsers just call snd_hda_gen_free().

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 8a02c0cc4a5915e01bf19ef030166567713d0eb7)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_generic.c b/sound/pci/hda/hda_generic.c
index a9b3677..120a40f 100644
--- a/sound/pci/hda/hda_generic.c
+++ b/sound/pci/hda/hda_generic.c
@@ -5350,6 +5350,7 @@ EXPORT_SYMBOL_GPL(snd_hda_gen_init);
  */
 void snd_hda_gen_free(struct hda_codec *codec)
 {
+ snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_FREE);
  snd_hda_detach_beep_device(codec);
  snd_hda_gen_spec_free(codec->spec);
  kfree(codec->spec);
diff --git a/sound/pci/hda/patch_conexant.c b/sound/pci/hda/patch_conexant.c
index b474c7f..59e3aea 100644
--- a/sound/pci/hda/patch_conexant.c
+++ b/sound/pci/hda/patch_conexant.c
@@ -2798,11 +2798,7 @@ static int cx_auto_init(struct hda_codec *codec)
  return 0;
 }
 
-static void cx_auto_free(struct hda_codec *codec)
-{
- snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_FREE);
- snd_hda_gen_free(codec);
-}
+#define cx_auto_free snd_hda_gen_free
 
 static const struct hda_codec_ops cx_auto_patch_ops = {
  .build_controls = cx_auto_build_controls,
diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 7a628e8..d177ffd 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -845,11 +845,7 @@ static inline void alc_shutup(struct hda_codec *codec)
   snd_hda_shutup_pins(codec);
 }
 
-static void alc_free(struct hda_codec *codec)
-{
- snd_hda_apply_fixup(codec, HDA_FIXUP_ACT_FREE);
- snd_hda_gen_free(codec);
-}
+#define alc_free snd_hda_gen_free
 
 #ifdef CONFIG_PM
 static void alc_power_eapd(struct hda_codec *codec)
-- 
1.7.1